### ISSUE TITLE
IClient connection status.

### ISSUE THREAD

**Issue by Nucs** [mathpaquette/IQFeed.CSharpApiClient#25]
There is no way to know if a client is connected, disconnected or suddenly lost connection.
There is a heavy need for:
* Event for connection state change.
* Property for current connection state.

**Comment by mathpaquette** [comment]
@Nucs did you check with AdminClient?

**Comment by Nucs** [comment]
I'm talking about knowing in my business layer if `Level1Client` has disconnected, lost connection or just idle because the specific stock is silent.

You'll need to be more specific because there is no documentation at all in any parts of the code.
`ClientsConnected` from `StatsMessage` in `AdminClient` provides general count of connections.
`ClientStatsMessage` still doesn't refer to `Level1Client` current state.

**Comment by mathpaquette** [comment]
I didnt forget you... Still working to find the time to add this. Before the end of the week for sure!

**Comment by mathpaquette** [comment]
Nucs, Level1 you will receive Timestamp every seconds, if not, something goes wrong.

**Comment by hunterfries** [comment]
Wanted to circle back to this...what about checking the status of the DerivativeClient? It is also streaming data and needs to handle connection state changes.

**Comment by hunterfries** [comment]
Bump here...what do we think about this?

**Comment by mathpaquette** [comment]
@hunterfries totally agree with you guys that would be nice to have such abiliality in every connected clients. I'll do it once I have spare time, right now focusing on my new contract...

**Comment by hunterfries** [comment]
No problem. In the meantime, would subscribing to the Level1Client's timestamp callback and then handling a disconnection be the current workaround? Something like: 

 IQFeedLauncher.Terminate();
IQFeedLauncher.Start();
// start your clients

@Nucs is this how your handling this?

**Comment by Nucs** [comment]
The same way I wrote on the first post on a separate thread.

**Comment by mathpaquette** [comment]
But guys, do you agree that if there's no reconnect mechanism checking the status is a but useless or this is me that not getting the use case?

**Comment by hunterfries** [comment]
That or an implementation of the workaround above that fired a Disconnected event that we could hook onto. I would just want some input on the implementation.

**Comment by mathpaquette** [comment]
@hunterfries, yes, I think I prefer the Disconnected event. Ill do that. thanks

**Comment by 10thBeast** [comment]
Hi All, 

What is the **Product_id** in app.config file or user environment.

**Comment by mathpaquette** [comment]
@10thBeast long story short, you need to buy a dev license from IQFeed

**Comment by mathpaquette** [comment]
Guys,  there's no easy way to know if a socket is still connected or not. Even the Connected property from Socket class can lie about his current status in the case the connection dropped unexpectedly. We need to have a HeartBeat mechanism but not supported by IQFeed... 

Ill check what I can do with the Admin port to monitor the status with IQFeed Servers but there's no easy way to do it.

**Comment by Nucs** [comment]
You can always create a seperate `Level1Client `that will use timestamps as heartbeat while a different client is used for data.
